File: client/java/src/avt/Part.java
package avt;

import javax.microedition.lcdui.Graphics;

public abstract class Part {
   public short follow;
   public short IDPart;
   public short idIcon;
   public int[] price = new int[2];
   public byte zOrder;
   public byte sell;
   public String name;

   public final void paint(Graphics var1, int var2, int var3, int var4) {
      if (this.IDPart != -1) {
         if (this.IDPart >= 2000) {
            boolean var7 = false;
            short var9 = this.idIcon;
            ImageIcon var10;
            if ((var10 = AvatarData.getImagePart(var9)).count != -1 || this.IDPart == -1) {
               var1.drawRegion(var10.img, 0, 0, var10.w, var10.h, 0, var2, var3, var4);
            }

            return;
         }

         AvatarData.listImgInfo[this.idIcon].paintPart(var1, var2, var3, var4);
      }

   }

   public void paintIcon(Graphics var1, int var2, int var3, int var4, int var5) {
      this.paint(var1, var2, var3, var5);
   }

   public void paintAvatar(Graphics var1, int var2, int var3, int var4, int var5) {
   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/Part.cs
public abstract class Part
{
	public short follow;

	public short IDPart;

	public short idIcon;

	public int[] price = new int[2];

	public sbyte zOrder;

	public sbyte sell;

	public string name;

	public virtual void paint(MyGraphics g, int x, int y, int arthor)
	{
		if (IDPart != -1)
		{
			if (IDPart >= 2000)
			{
				paintDynamic(g, idIcon, x, y, 0, arthor);
			}
			else
			{
				AvatarData.listImgInfo[idIcon].paintPart(g, x, y, arthor);
			}
		}
	}

	public virtual void paintIcon(MyGraphics g, int x, int y, int direct, int arthor)
	{
		paint(g, x, y, arthor);
	}

	public virtual void paintAvatar(MyGraphics g, int index, int x, int y, int direct, int arthor)
	{
	}

	public void paintDynamic(MyGraphics g, short idImg, int x, int y, int direct, int arthor)
	{
		ImageIcon imagePart = AvatarData.getImagePart(idImg);
		if (imagePart.count != -1 || IDPart == -1)
		{
			g.drawRegion(imagePart.img, 0f, 0f, imagePart.w, imagePart.h, direct, x, y, arthor);
		}
	}
}
